src/common/hostProcess.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @name - Host Process
+ * @description - Describes the host process
+ * @member type - The type of the host process (electron, browser, etc)
+ * @member release - The release string of the host process
+ */
+export interface IHostProcess {
+    type: HostProcessType;
+    release: string;
+}
+
+/**
+ * @enum ELECTRON - Electron Host Process Type
+ * @enum BROWSER - Browser Host Process Type
+ */
+export enum HostProcessType {
```

Comment:
I recommend we use simplier values to make this more clear and use power of 2 example from https://patrickdesjardins.com/blog/how-to-set-and-read-bitwise-enum-values-in-typescript

Reply:
IMHO I think writing out the bit-shifting is more self explanatory than writing the decimal representations (1, 2, 3). With the decimal form, I think someone unfamiliar with what's happening could easily come in and put a new entry with an incorrect number, for example:
```
enum HostProcessType {
  Electron = 1, // 001
  Browser = 2, // 010
  NEW = 3,      // 011
  All = 4,          // 100
}
```
But in reality the correct numbering should be: 1, 2, 4, 7.

Reply:
chatted offline, and I also agree the decimal for with better comments is more understandable. The set all bits syntax is a bit cryptic.